### PR TITLE
Update parameter defaults for ``QuotaClient.update_quota()``

### PR DESCRIPTION
**Pull request by bernt-matthias**

### CODE REVIEW

File: bioblend/_tests/TestGalaxyQuotas.py [hunk 1]
```diff
@@ -53,13 +56,21 @@ def test_delete_undelete_quota(self):
         response = self.gi.quotas.undelete_quota(self.quota["id"])
         assert response == "Undeleted 1 quotas: " + self.quota_name
 
-
+    @test_util.skip_unless_galaxy("release_19.09")  # for user purging
     def test_update_non_default_quote(self):
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyQuotas.py:60]:
```suggestion
    def test_update_non_default_quota(self):
```

File: bioblend/_tests/TestGalaxyQuotas.py [hunk 2]
```diff
@@ -52,3 +55,22 @@ def test_delete_undelete_quota(self):
...
+        quota = self.gi.quotas.create_quota(name="non_default_quota", description="testing",
+                amount="100 GB",
+                operation="+",
+                in_users=[new_user['username']],
+            )
+        self.gi.quotas.update_quota(quota["id"], default="no", amount="200 GB")
+
+        if self.gi.config.get_config()["allow_user_deletion"]:
+            deleted_user = self.gi.users.delete_user(new_user["id"])
+            purged_user = self.gi.users.delete_user(new_user["id"], purge=True)
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyQuotas.py:76]:
```suggestion
            self.gi.users.delete_user(new_user["id"])
            self.gi.users.delete_user(new_user["id"], purge=True)
```

File: bioblend/_tests/TestGalaxyQuotas.py [hunk 3]
```diff
@@ -52,3 +55,22 @@ def test_delete_undelete_quota(self):
...
+        new_username = test_util.random_string()
+        new_user_email = f"{new_username}@example.org"
+        password = test_util.random_string(20)
+        new_user = self.gi.users.create_local_user(new_username, new_user_email, password)
+        print(new_user)
+
+        quota = self.gi.quotas.create_quota(name="non_default_quota", description="testing",
+                amount="100 GB",
+                operation="+",
+                in_users=[new_user['username']],
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyQuotas.py:70]: I think this needs to be either user ids or email addresses, see https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/services/quotas.py#L126

Comment (bernt-matthias) [reply bioblend/_tests/TestGalaxyQuotas.py]: Indeed

**Comment by nsoranzo** [comment]
Pushed to the main branch a fix for:
```
bioblend/galaxy/workflows/__init__.py:21: error: Unused "type: ignore" comment 
[unused-ignore]
    InputsBy = Literal["step_index|step_uuid", "step_index", "step_id", "s...
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
```

**Comment by bernt-matthias** [comment]
Thanks @nsoranzo . The remaining linter error seems to depend on the answers to the above question(s).

**Comment by nsoranzo** [comment]
@bernt-matthias Are you happy with the changes I pushed? They align with what the Galaxy API currently does.

File: bioblend/_tests/TestGalaxyQuotas.py [hunk 4]
```diff
@@ -52,3 +55,29 @@ def test_delete_undelete_quota(self):
...
+        new_user = self.gi.users.create_local_user(new_username, new_user_email, password)
+
+        quota = self.gi.quotas.create_quota(
+            name="non_default_quota",
+            description="testing",
+            amount="100 GB",
+            operation="+",
+            in_users=[new_user["id"]],
+        )
+        self.gi.quotas.update_quota(quota["id"], default=None, amount="200 GB")
```
Comment (bernt-matthias) [inline bioblend/_tests/TestGalaxyQuotas.py:79]: should we test like this?

```suggestion
        self.gi.quotas.update_quota(quota["id"], amount="200 GB")
```

File: bioblend/_tests/TestGalaxyQuotas.py [hunk 5]
```diff
@@ -52,3 +55,29 @@ def test_delete_undelete_quota(self):
         assert response == "Deleted 1 quotas: " + self.quota_name
         response = self.gi.quotas.undelete_quota(self.quota["id"])
         assert response == "Undeleted 1 quotas: " + self.quota_name
+
+    @test_util.skip_unless_galaxy("release_19.09")  # for user purging
+    def test_update_non_default_quota(self):
+        """
+        test if a non default quota can be updated (needs to use default=None,
+        default="no" will fail)
```
Comment (bernt-matthias) [inline bioblend/_tests/TestGalaxyQuotas.py:63]:
```suggestion
        test if a non default quota can be updated. needs to use `default=None` (which is the default), `default="no"` will fail
```

File: bioblend/_tests/TestGalaxyQuotas.py (hunk 4 above)
Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyQuotas.py:79]: I don't mind either way

**Review by bernt-matthias — COMMENTED** [review]
Thanks a lot. Lets go :)

**Pull request merged**